client/post-editor/editor-title/index.jsx

```diff
@@ -101,19 +104,36 @@ export default React.createClass( {
 				}
 				<TrackInputChanges onNewValue={ this.recordChangeStats }>
 					<TextareaAutosize
-						{ ...omit( this.props, Object.keys( this.constructor.propTypes ) ) }
+						{ ...ownProps }
 						className="editor-title__input"
-						placeholder={ this.translate( 'Title' ) }
+						placeholder={ translate( 'Title' ) }
 						onChange={ this.onChange }
 						onInput={ this.resizeAfterNewlineInput }
 						onBlur={ this.onBlur }
 						autoFocus={ isNew && ! isMobile() }
 						value={ post ? post.title : '' }
-						aria-label={ this.translate( 'Edit title' ) }
+						aria-label={ translate( 'Edit title' ) }
 						ref="titleInput"
 						rows="1" />
 				</TrackInputChanges>
 			</div>
 		);
 	}
-} );
+}
+
+export default connect(
+	( state, ownProps ) => {
+		const site = getSelectedSite( state );
+		const editedPostId = getEditorPostId( state );
+		const post = getEditedPost( state, site.ID, editedPostId );
+		const isNew = isEditorNewPost( state );
+
+		return {
+			isNew,
+			post,
+			site,
+			ownProps
```

Comment:
I think I understand what you are saying now @youknowriad. it seems like an indication that bigger issues exist here.

we have plenty of places where we pass `...props` down through several levels of nesting and that is almost always really confusing. I'm a huge advocate for explicit prop passing. props passed by localize shouldn't be passed along the chain either because if a child needs `translate` then it should be directly wrapped by `localize`

concerning passing other props I wonder if it's not more of a concern than it seems. why should we want to pass some new style to an underlying `textarea` inside a component? that actually breaks our React idioms, that parents should need to know nothing of the inner workings of their children. perhaps being explicit about a component's public API and keeping the nested children isolated from the grandparents can provide more robustness and reliability here.